acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/GitCreationPage.java

```diff
@@ -79,4 +70,30 @@ public MultiBranchPipeline createPipeline(SSEClientRule sseCLient, String pipeli
         pipeline.getActivityPage().checkUrl();
         return pipeline;
     }
+
+    public MultiBranchPipeline createPipelinePW(SSEClientRule sseCLient, String pipelineName, String url, String user, String pass) throws IOException {
+        jobApi.deletePipeline(pipelineName);
+        dashboardPage.clickNewPipelineBtn();
+        clickGitCreationOption();
+        wait.until(By.cssSelector("div.text-repository-url input")).sendKeys(url);
+
+        wait.until(By.xpath("//*[contains(text(), 'Jenkins needs a user credential')]"));
+
+        wait.until(By.cssSelector("div.text-username input")).sendKeys(user);
+        wait.until(By.cssSelector("div.text-password input")).sendKeys(pass);
+        wait.until(By.cssSelector(".button-create-credential")).click();
+
+        wait.until(By.xpath("//*[contains(text(), 'Use existing credential')]"));
```

Comment:
I know this comment is in the test code, and not the UI code, but I'll ask the question here and hope it makes sense. Any chance we could have a cssSelector here? That way, if the text in messages.properties gets changed for some reason, the test will still work.

Reply:
I really don't like adding classes just for test cases, as every time we do it we add a tiny but persistent and ever-increasing bloat to the code and maintenance load (and browser performance). But mostly because that's not what CSS classnames are for.

I'll change it if there's some sort of consensus, but it ain't right agnabbit.